odoo/__init__.py

```diff
@@ -85,6 +85,14 @@ def _decompress(data):
 except ImportError:
     pass # no fix required
 
+#----------------------------------------------------------
+# some charset are known by Python under a different name
+#----------------------------------------------------------
+import encodings.aliases
+
+encodings.aliases.aliases['874'] = 'cp874'
+encodings.aliases.aliases['windows_874'] = 'cp874'
+encodings.aliases.aliases['windows-874'] = 'cp874'
```

Comment:
No need to add an alias for `windows-874`. You can register onle `windows_874` and `windows-874` will work out of the box. https://github.com/python/cpython/blob/15309329b65a285cb7b3071f0f08ac964b61411b/Lib/encodings/__init__.py#L43-L69